### PR TITLE
feat: add `collectFromComposedSchemas()` and `getExamplesForSchema()` to utilities

### PR DESCRIPTION
**Pull request by hudlow**
## PR summary
Adds a generic `collectFromComposedSchemas()` function that can be used to build functions like the also-added `getExamplesForSchema()`, and refactors `getPropertyNamesForSchema()` to use `collectFromComposedSchemas()` as well.

### General checklist
Please make sure that your PR fulfills the following requirements:
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [x] `.secrets.baseline` has been updated as needed
- [x] `npm run generate-utilities-docs` has been run if any files in `packages/utilities/src` have been updated

### CODE REVIEW

File: packages/utilities/src/utils/collect-from-composed-schemas.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2017 - 2023 IBM Corporation.
```
Comment (dpopp07) [inline packages/utilities/src/utils/collect-from-composed-schemas.js:2]:
```suggestion
 * Copyright 2024 IBM Corporation.
```

(Not the only instance of this)

File: packages/utilities/src/utils/collect-from-composed-schemas.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+/**
+ * Returns an array of items collected by the provided `collector(schema) => item[]` function for a
+ * simple or composite schema, and deduplicate primitives in the result. The collector function is
```
Comment (dpopp07) [inline packages/utilities/src/utils/collect-from-composed-schemas.js:8]:
```suggestion
 * simple or composite schema, and deduplicates primitives in the result. The collector function is
```

File: packages/utilities/src/utils/get-examples-for-schema.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Returns an array of examples for a simple or composite schema. For each composed schema, if
+ * `schema.examples` is present (and an array), `schema.example` is ignored.
+ * @param {object} schema simple or composite OpenAPI 3.0 schema object
+ * @returns {Array} examples
+ */
+function getExamplesForSchema(schema) {
+  return collectFromComposedSchemas(schema, s => {
+    if (Array.isArray(s.examples)) {
```
Comment (dpopp07) [inline packages/utilities/src/utils/get-examples-for-schema.js:19]: I'm under the impression that `examples` is only supported on Parameter and Media Type objects, not on schemas generally, and that it's a map. I thought only `example` was supported on schemas - does the spec allow for a list of examples as well?

File: packages/utilities/test/get-examples-for-schema.test.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  it('should return `[]` for empty object', async () => {
+    expect(getExamplesForSchema({})).toEqual([]);
+  });
+
+  it('should return examples for simple schema', async () => {
+    expect(
+      getExamplesForSchema({
+        examples: ['three', 'two', 'one'],
+      }).sort()
+    ).toEqual(['three', 'two', 'one'].sort());
```
Comment (dpopp07) [inline packages/utilities/test/get-examples-for-schema.test.js:26]: I wouldn't think that `sort()` would be necessary for a list of primitives - am I missing something?

File: packages/utilities/test/get-examples-for-schema.test.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+  });
+
+  it('should return single example for simple schema', async () => {
+    expect(
+      getExamplesForSchema({
+        example: 'one',
+      }).sort()
```
Comment (dpopp07) [inline packages/utilities/test/get-examples-for-schema.test.js:33]: Or for lists of size one 🙂

**Review by dpopp07 — COMMENTED** [review]
First round - mostly just typos and questions

File: packages/utilities/src/utils/get-examples-for-schema.js (hunk 3 above)
Comment (hudlow) [reply packages/utilities/src/utils/get-examples-for-schema.js:19]: OpenAPI 3.1 [deprecates `example`](https://spec.openapis.org/oas/v3.1.0#fixed-fields-19) in favor of ["native" JSON Schema `examples`](https://json-schema.org/draft/2020-12/json-schema-validation#name-examples).

File: packages/utilities/test/get-examples-for-schema.test.js (hunk 4 above)
Comment (hudlow) [reply packages/utilities/test/get-examples-for-schema.test.js:26]: Technically, there are no guarantees on the order examples are returned in.

File: packages/utilities/src/utils/get-examples-for-schema.js (hunk 3 above)
Comment (dpopp07) [reply packages/utilities/src/utils/get-examples-for-schema.js:19]: Wow, I'm not sure how I got mixed up there - I thought `example` was the native field from JSON Schema. I probably need to update some things in my new rule based on that knowledge, as well. Thanks for linking that!

**Review by dpopp07 — APPROVED** [review]
Looks great! 👍 

The only other thing I wondered about is if we'd want to return a map with the examples, instead of a list, with the actual paths attached somehow - but that would require some rework of the logic and I'm not sure it's worth it. Rules utilizing this should already be able to know the logical path, which would be more useful. So, I think this is ready to merge as-is.

**Pull request merged**